proxy/proxy_test.go

```diff
@@ -845,25 +846,25 @@ func TestHostHeader(t *testing.T) {
 	}, {
 		"no proxy preserve, route preserve not, explicit host last",
 		FlagsNone,
-		`route: Any() -> preserveHost("false") -> requestHeader("Host", "custom.example.org") -> "%s"`,
+		`route: Any() -> preserveHost("false") -> appendRequestHeader("Host", "custom.example.org") -> "%s"`,
```

Comment:
I just followed the code and replaced the deprecated filter with the current replacement with similar code. We can change that if it makes sense. :)

Reply:
let me push it

Reply:
pushed this. I couldn't fully test the project for some reason, because currently on my local the proxy package tests take forever, also on master. Checking why.